collect_app/src/main/java/org/odk/collect/android/activities/FormFillingActivity.java

```diff
@@ -2016,6 +2016,7 @@ && new PlayServicesChecker().isGooglePlayServicesAvailable(this)) {
                                 formControllerAvailable(formController, form, instance);
                                 Intent intent = new Intent(this, FormHierarchyFragmentHostActivity.class);
                                 intent.putExtra(FormHierarchyFragmentHostActivity.EXTRA_SESSION_ID, sessionId);
+                                intent.putExtra(FormHierarchyFragmentHostActivity.SHOW_NEW_EDIT_MESSAGE, formEntryViewModel.shouldShowNewEditMessage());
```

Comment:
Ah this definitely more annoying because of the Activity split here. Still, I think it could be a little simpler. Ideally, we could just show the message on form load (if the hierarchy was part of the same Activity) based on the `shouldShowNewEditMessage` return value. I think we can achieve something similar but just showing the snackbar in `FormHierarchyFragmentHostActivity` in `onCreate` based on this extra. It would show again if you rotated the hierarchy, but only on that initial load, so I don't think it's a big deal.

Reply:
Done.